integration-test.Rmd

```diff
@@ -26,15 +26,18 @@ knitr::opts_chunk$set(
 Packages used in this file:
 
 ```{r}
-suppressPackageStartupMessages(library(tidyverse))
-suppressPackageStartupMessages(library(devtools))
-suppressPackageStartupMessages(library(testthat))
-suppressPackageStartupMessages(library(config))
-suppressPackageStartupMessages(library(rlang))
-suppressPackageStartupMessages(library(renv))
-suppressPackageStartupMessages(library(glue))
-suppressPackageStartupMessages(library(fs))
+library(tidyverse)
+library(devtools)
+library(testthat)
+library(config)
+library(rlang)
+library(renv)
+library(glue)
+library(fs)
 library(here)
+library(conflicted)
+conflicted::conflict_prefer("filter", "dplyr")
+conflicted::conflict_prefer("lag", "dplyr")
```

Comment:
Getting some help from the package "conflicted". Eventually I would prune to a minimum the packages we depend on.

Reply:
sounds good and agreed. I will take a look into best practices on how to minimise dependencies and check which ones best to remove

Reply:
I think @cjyetman once lead a ds-incubator on the topic?
A good place to get some ideas is https://www.tidyverse.org/blog/2019/05/itdepends/

Reply:
very much in favor of reducing dependencies... and thereby reducing conflicts, and reducing package version issues, and reducing system setup time/complexity, and reducing run time, and reducing other random problems 😄

also be aware that the order in which packages are loaded will affect which conflicting functions take precedence... and when you run multiple scripts and/or do not reset your session, what's already loaded in your environment can change what the script you're looking at appears to suggest as the load order (e.g. the last package to load may have already been loaded by a previous script)

`tidyverse` is an obvious one to try to get rid of.... I believe that `library(tidyverse)` is intended as a shortcut for interactive use, but it's probably much safer in a script to only load the specific packages that are needed from `tidyverse`